src/main/java/core/GraphReducer.java

```diff
@@ -61,12 +68,34 @@ public static void determineParents(HashMap<Integer, Node> nodeMap) {
     }
 
     /**
+     * Copy all values of a given node map.
+     *
+     * @param map A node map to be copied.
+     * @return A copied node map.
+     */
+    private static HashMap<Integer, Node> copyNodeMap(HashMap<Integer, Node> map) {
+        HashMap<Integer, Node> res = new HashMap<Integer, Node>();
+        for (int i : map.keySet()) {
+            Node n = map.get(i);
+            Node newNode = new Node(n.getId(), n.getSequence(), n.getzIndex());
+            newNode.setLinks(n.getLinks());
+            newNode.setParents(n.getParents());
+            newNode.setGenomes(n.getGenomes());
+            res.put(i, newNode);
+        }
+
+        return res;
+    }
```

Comment:
No, in that case only a shallow copy will be made and level map 0 will have collapsed node, something that should happen.